zmq.hpp

```diff
@@ -1284,6 +1284,7 @@ namespace sockopt
 // BoolUnit: if true accepts values of type bool (but passed as T into libzmq)
 template<int Opt, class T, bool BoolUnit = false> struct integral_option
 {
+    ZMQ_CONSTEXPR_VAR integral_option(){}
```

Comment:
This can be `constexpr integral_option() noexcept = default;`, same applies to array_option.

Reply:
Thanks for the review and suggestion. I will try and report back.

Reply:
It turns out that your suggestion also leads to the following build error on macOS 10.11.6:

```
/path/to/cppzmq/zmq.hpp:1329:39: error: default initialization of an object of const type 'const affinity_t' (aka 'const integral_option<4, unsigned long long, false>') without a user-provided default constructor
ZMQ_DEFINE_INTEGRAL_OPT(ZMQ_AFFINITY, affinity, uint64_t);
                                      ^
```

Reply:
Let me know if you need me to test anything else to move forward with the integration :pray:

Reply:
OK, I looked into this and I think the best solution is to change the definition macros to `ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME{}` (added {}).

Reply:
Thanks, I will update the PR later this afternoon

Reply:
This last suggestion also lead to the same error on macOS 10.11.6.

Here is the patch I applied:

```diff
@@ -1299,10 +1301,10 @@ template<int Opt, int NullTerm = 1> struct array_option
     ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME
 #define ZMQ_DEFINE_INTEGRAL_BOOL_UNIT_OPT(OPT, NAME, TYPE)                          \
     using NAME##_t = integral_option<OPT, TYPE, true>;                              \
-    ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME
+    ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME{}
 #define ZMQ_DEFINE_ARRAY_OPT(OPT, NAME)                                             \
     using NAME##_t = array_option<OPT>;                                             \
-    ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME
+    ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME{}
 #define ZMQ_DEFINE_ARRAY_OPT_BINARY(OPT, NAME)                                      \
     using NAME##_t = array_option<OPT, 0>;                                          \
     ZMQ_INLINE_VAR ZMQ_CONSTEXPR_VAR NAME##_t NAME
```

Reply:
Hmm, did you apply this change to all 5 macros or just those 2?

Reply:
Updating all for macro addressed the issue. I just forced push the topic.